packages/dds/tree/src/simple-tree/api/tree.ts

```diff
@@ -359,6 +359,89 @@ export function checkUnion(union: Iterable<TreeNodeSchema>, errors: string[]): v
 	}
 }
 
+/**
+ * A context associated with a (version-control style) branch of a shared tree.
+ * @remarks A `TreeContext` allows for the {@link TreeContext.branch | creation of branches} and for those branches to later be {@link TreeContext.merge | merged}.
+ * The `TreeContext` for a specific {@link TreeNode} may be acquired by calling `TreeAlpha.context`.
+ * A context does not necessarily know the schema of its SharedTree - to convert a context to a {@link TreeViewAlpha | view with a schema}, use {@link TreeContext.hasRootSchema | hasRootSchema()}.
+ * @sealed @alpha
+ */
+export interface TreeContext {
```

Comment:
Why we name it `TreeContext` instead of `TreeBranch`?

Reply:
I feel like it's actually a good thing to name something so fundamental in a way that it is associated with branching, as having a branching system is one of our key differentiators and something we really want users to take advantage of.

Reply:
It also lets us dodge the problem of what to rename `isBranch()`, if `TreeContext` is a `TreeBranch`

Reply:
Nah you're right.